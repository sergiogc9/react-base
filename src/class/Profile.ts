export interface ProfileObject {}

export class Profile {

	// public user: User;

	// constructor(profile: ProfileObject) {
	// 	this.user = new User(profile.user);
	// }
}
