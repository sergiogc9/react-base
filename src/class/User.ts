export class UserObject {
	public id: string;
	public name: string;

	constructor(values?: Partial<UserObject>) {
		if (values) Object.assign(this, values);
	}
}

export class User extends UserObject {

}
