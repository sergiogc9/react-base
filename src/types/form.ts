import { TextFieldProps } from "react-md/lib/TextFields";
import { SelectFieldProps } from "react-md/lib/SelectFields/SelectField";
import { AutocompleteProps } from "react-md/lib/Autocompletes/Autocomplete";
import { DatePickerProps } from "react-md/lib/Pickers";
import { ButtonProps } from "react-md/lib/Buttons";
import { AddFileError } from "@src/components/common/Dropzone/types";

export type FormFieldItem = { label: string, value: string };

export type BaseFormValidations = {
    fn?: (value: any) => boolean
};
type BaseFormStyle = {
    width?: 25 | 33 | 50 | 66 | 75 | 100,
    newLine?: boolean, // Forces field to be put in a new line (even if it should fit in current line)
    uniqueInLine?: boolean // Forces field to be put only itself in the line
}

export type BaseFieldElement = {
    id: string,
    required?: boolean,
    disabled?: boolean,
    className?: string,
    style?: BaseFormStyle,
    onValueUpdated?: (value: any) => void
};

export type TextFieldElementValidations = BaseFormValidations & { maxLength?: number, minLength?: number, url?: boolean };

export type TextFieldElement = BaseFieldElement & {
    type: "text",
    label: string | { key: string },
    defaultValue?: string,
    inputProps?: Partial<TextFieldProps>,
    validations?: TextFieldElementValidations
};

export type NumberFieldElementValidations = BaseFormValidations & { max?: number, min?: number };

export type NumberFieldElement = BaseFieldElement & {
    type: "number",
    label: string | { key: string },
    defaultValue?: number,
    inputProps?: Partial<TextFieldProps>,
    validations?: NumberFieldElementValidations
};

export type SelectFieldElementValidations = BaseFormValidations;

export type SelectFieldElement = BaseFieldElement & {
    type: "select",
    label: string | { key: string },
    items: FormFieldItem[],
    defaultValue?: string,
    inputProps?: Partial<SelectFieldProps>,
    validations?: SelectFieldElementValidations
};

export type AutocompleteFieldElementValidations = BaseFormValidations & {
    url?: boolean
};

export type AutocompleteFieldElement = BaseFieldElement & {
    type: "autocomplete",
    label: string | { key: string },
    items: FormFieldItem[],
    defaultValue?: FormFieldItem,
    inputProps?: Partial<AutocompleteProps>,
    validations?: AutocompleteFieldElementValidations,
    closeButton?: boolean,
    useValueFromAutocomplete?: boolean, // Uses item.value from result instead of item.label as the new label
    alwaysShowItems?: boolean, // Show items even if no filtered value is set
    onUpdateItems?: (newValue: string) => Promise<AutocompleteFieldElement["items"]>
};

export type FileFieldElementValidations = BaseFormValidations;

export type FileFieldElement = BaseFieldElement & {
    type: "file",
    defaultValue?: File[],
    validations?: FileFieldElementValidations,
    multiple: boolean,
    accept?: string,
    maxSize?: number,
    onFileErrors?: (errors: AddFileError[]) => void // Some default errors are yet catched by Field component itself
};

export type DateFieldElementValidations = BaseFormValidations;

export type DateFieldElement = BaseFieldElement & {
    type: "date",
    defaultValue?: Date,
    validations?: DateFieldElementValidations,
    minDate?: Date,
    maxDate?: Date,
    inputProps?: Partial<DatePickerProps>
};

export type CustomComponentField = {
    id: string,
    type: "component",
    component: JSX.Element,
    style?: BaseFormStyle,
};

type ButtonGroupOption = {
    id: string,
    label: string,
    value: string,
    buttonProps?: Partial<ButtonProps>
};

export type ButtonGroupFieldElementValidations = BaseFormValidations;

export type ButtonGroupFieldElement = BaseFieldElement & {
    type: "buttonGroup",
    options: ButtonGroupOption[],
    label?: string,
    defaultValue?: string,
    validations?: ButtonGroupFieldElementValidations
};

// eslint-disable-next-line max-len
export type FormFieldElement = TextFieldElement | NumberFieldElement | SelectFieldElement | AutocompleteFieldElement | FileFieldElement | DateFieldElement | CustomComponentField | ButtonGroupFieldElement;
