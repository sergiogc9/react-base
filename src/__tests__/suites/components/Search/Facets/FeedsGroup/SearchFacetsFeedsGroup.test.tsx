import { shallow } from "enzyme";
import React from "react";
import SearchFacetsFeedsGroup from "@src/components/Search/Facets/FeedsGroup/SearchFacetsFeedsGroup";
import GA from '@src/lib/googleAnalytics';
import { FeedObject } from "@src/class/Feed";
import { FocusFeeds, FocusObject } from "@src/class/Focus";
import { FocusFeedsGroupKey } from "@src/types/search/form";
import TestHelper from '@src/__tests__/utils/Helper/TestHelper';

const document = TestHelper.getDocument();

const feedOnline: FeedObject = {
	definition: {},
	deleted_at: new Date(1549287550563),
	enabled: true,
	filters: {},
	focus_id: "focus-id-1",
	id: "feed-id-1",
	inserted_at: new Date(1549287550563),
	name: "feed-name-1",
	oldest_document: document,
	type: "online",
	updated_at: new Date(1549287550563)
};

const feedSocial: FeedObject = {
	...feedOnline,
	id: "feed-id-2",
	name: "feed-name-2",
	type: "socialmedia",
};

const feedPrint: FeedObject = {
	...feedOnline,
	id: "feed-id-3",
	name: "feed-name-3",
	type: "print",
};

const focusFeeds: FocusFeeds = {
	online: [feedOnline],
	socialmedia: [feedSocial],
	print: [feedPrint]
};

const focus: FocusObject = {
	deleted_at: new Date(1549287550563),
	feeds: focusFeeds,
	id: "focus-id-1",
	inserted_at: new Date(1549287550563),
	name: "focus-name-1",
	oldest_document: document,
	updated_at: new Date(1549287550563),
	url_logo: "http://fake-url.com",
};

const focusList: FocusObject[] = [focus];

let onToggleFacetGroupOpened: any;
let onToggleFeedFiltered: any;
let onToggleFocusFiltered: any;
let onRemoveFocusFeedFilter: any;

describe("<SearchFacetsFeedsGroup />", () => {
	let wrapper: any;

	beforeAll(() => {
		window.matchMedia = jest.fn((query: string) => ({
			matches: query.indexOf('(min-width: 1025px)') !== -1,
		}));
	});

	beforeEach(() => {
		onToggleFacetGroupOpened = jest.fn();
		onToggleFeedFiltered = jest.fn();
		onToggleFocusFiltered = jest.fn();
		onRemoveFocusFeedFilter = jest.fn();
		wrapper = shallow(<SearchFacetsFeedsGroup
			focusList={focusList}
			focusFiltered={[]}
			feedsFiltered={[]}
			facetsGroupsOpened={[FocusFeedsGroupKey]}
			onToggleFacetGroupOpened={onToggleFacetGroupOpened}
			onToggleFeedFiltered={onToggleFeedFiltered}
			onToggleFocusFiltered={onToggleFocusFiltered}
			onRemoveFocusFeedFilter={onRemoveFocusFeedFilter}
		/>);
	});

	it("Test component snapshot with focus list not loaded", () => {
		wrapper.setProps({ focusList: null });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test component snapshot with group closed", () => {
		wrapper.setProps({ facetsGroupsOpened: [] });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test component snapshot with group opened and no filters enabled", () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test component snapshot with group disabled (no focus)", () => {
		wrapper.setProps({ focusList: [] });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test component snapshot with focus filtered", () => {
		wrapper.setProps({ focusFiltered: [focus.id] });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test component snapshot with a feed filtered", () => {
		wrapper.setProps({ feedsFiltered: [feedOnline.id] });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test toggle focus filtered checkbox handler", () => {
		GA.trackEvent = jest.fn();
		expect(onToggleFocusFiltered).toHaveBeenCalledTimes(0);
		expect(GA.trackEvent).toHaveBeenCalledTimes(0);
		wrapper.find(`#searchFacetsFeedsGroupItemCheckbox-focus-${focus.id}`).at(0).props().onChange();
		expect(onToggleFocusFiltered).toHaveBeenCalledTimes(1);
		expect(GA.trackEvent).toHaveBeenCalledTimes(1);
		wrapper.find(`#searchFacetsFeedsGroupItemName-focus-${focus.id}`).at(0).simulate('click');
		expect(onToggleFocusFiltered).toHaveBeenCalledTimes(2);
		expect(GA.trackEvent).toHaveBeenCalledTimes(2);
	});

	it("Test toggle feed filtered checkbox handler", () => {
		GA.trackEvent = jest.fn();
		expect(onToggleFeedFiltered).toHaveBeenCalledTimes(0);
		expect(GA.trackEvent).toHaveBeenCalledTimes(0);
		wrapper.find(`#searchFacetsFeedsGroupItemCheckbox-feed-${feedOnline.id}`).at(0).props().onChange();
		expect(onToggleFeedFiltered).toHaveBeenCalledTimes(1);
		expect(GA.trackEvent).toHaveBeenCalledTimes(1);
		wrapper.find(`#searchFacetsFeedsGroupItemName-feed-${feedOnline.id}`).at(0).simulate('click');
		expect(onToggleFeedFiltered).toHaveBeenCalledTimes(2);
		expect(GA.trackEvent).toHaveBeenCalledTimes(2);
	});

	it("Test toggle group opened with group enabled (focus available)", () => {
		expect(onToggleFacetGroupOpened).toHaveBeenCalledTimes(0);
		wrapper.find(`#searchFacetsFeedsGroup-focus .search-facets-group-header`).at(0).simulate('click');
		expect(onToggleFacetGroupOpened).toHaveBeenCalledTimes(1);
	});

	it("Test toggle group opened with group disabled (no focus)", () => {
		wrapper.setProps({ focusList: [] });
		expect(onToggleFacetGroupOpened).toHaveBeenCalledTimes(0);
		wrapper.find(`#searchFacetsFeedsGroup-focus .search-facets-group-header`).at(0).simulate('click');
		expect(onToggleFacetGroupOpened).toHaveBeenCalledTimes(0);
	});

	it("Test remove all focus and feeds filtered", () => {
		wrapper.setProps({ focusFiltered: [focus.id] });
		expect(onRemoveFocusFeedFilter).toHaveBeenCalledTimes(0);
		wrapper.find(`#searchFacetsFeedsGroup-focus .search-facets-group-remove-filters`).at(0).simulate('click');
		expect(onRemoveFocusFeedFilter).toHaveBeenCalledTimes(1);
	});
});
