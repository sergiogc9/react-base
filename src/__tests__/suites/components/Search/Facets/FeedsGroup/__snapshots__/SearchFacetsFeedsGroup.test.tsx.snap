// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchFacetsFeedsGroup /> Test component snapshot with a feed filtered 1`] = `"<div class=\\"search-facets-group \\" id=\\"searchFacetsFeedsGroup-focus\\"><div class=\\"search-facets-group-header enabled\\"><i class=\\"md-icon material-icons\\">keyboard_arrow_down</i><span>filters.source.value</span><div class=\\"search-facets-group-remove-filters\\" title=\\"filters.reset.value\\"><i class=\\"md-icon material-icons\\">close</i></div></div><div class=\\"search-facets-group-items\\"><div class=\\"search-facets-group-item\\"><div class=\\"md-selection-control-container checkbox-green\\"><input type=\\"checkbox\\" id=\\"searchFacetsFeedsGroupItemCheckbox-focus-focus-id-1\\" class=\\"md-selection-control-input\\" name=\\"searchFacetsFeedsGroupItemCheckboxs-focus-focus-id-1[]\\" aria-hidden=\\"true\\"/><label for=\\"searchFacetsFeedsGroupItemCheckbox-focus-focus-id-1\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"false\\" aria-labelledby=\\"\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">indeterminate_check_box</i></div></label></div><div id=\\"searchFacetsFeedsGroupItemName-focus-focus-id-1\\" class=\\"search-facets-group-item-name\\">filters.source.focus: focus-name-1</div></div><div class=\\"search-facets-group-item feed-item\\"><div class=\\"md-selection-control-container checkbox-green\\"><input type=\\"checkbox\\" id=\\"searchFacetsFeedsGroupItemCheckbox-feed-feed-id-1\\" checked=\\"\\" class=\\"md-selection-control-input\\" name=\\"searchFacetsFeedsGroupItemCheckboxs-feed-feed-id-1[]\\" aria-hidden=\\"true\\"/><label for=\\"searchFacetsFeedsGroupItemCheckbox-feed-feed-id-1\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"true\\" aria-labelledby=\\"\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--theme-secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box</i></div></label></div><i class=\\"md-icon icon-news\\"></i><div id=\\"searchFacetsFeedsGroupItemName-feed-feed-id-1\\" class=\\"search-facets-group-item-name\\">feed-name-1</div></div><div class=\\"search-facets-group-item feed-item\\"><div class=\\"md-selection-control-container\\"><input type=\\"checkbox\\" id=\\"searchFacetsFeedsGroupItemCheckbox-feed-feed-id-2\\" class=\\"md-selection-control-input\\" name=\\"searchFacetsFeedsGroupItemCheckboxs-feed-feed-id-2[]\\" aria-hidden=\\"true\\"/><label for=\\"searchFacetsFeedsGroupItemCheckbox-feed-feed-id-2\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"false\\" aria-labelledby=\\"\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box_outline_blank</i></div></label></div><i class=\\"md-icon icon-socialmedia\\"></i><div id=\\"searchFacetsFeedsGroupItemName-feed-feed-id-2\\" class=\\"search-facets-group-item-name\\">feed-name-2</div></div><div class=\\"search-facets-group-item feed-item\\"><div class=\\"md-selection-control-container\\"><input type=\\"checkbox\\" id=\\"searchFacetsFeedsGroupItemCheckbox-feed-feed-id-3\\" class=\\"md-selection-control-input\\" name=\\"searchFacetsFeedsGroupItemCheckboxs-feed-feed-id-3[]\\" aria-hidden=\\"true\\"/><label for=\\"searchFacetsFeedsGroupItemCheckbox-feed-feed-id-3\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"false\\" aria-labelledby=\\"\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box_outline_blank</i></div></label></div><i class=\\"md-icon icon-print\\"></i><div id=\\"searchFacetsFeedsGroupItemName-feed-feed-id-3\\" class=\\"search-facets-group-item-name\\">feed-name-3</div></div></div></div>"`;

exports[`<SearchFacetsFeedsGroup /> Test component snapshot with focus filtered 1`] = `"<div class=\\"search-facets-group \\" id=\\"searchFacetsFeedsGroup-focus\\"><div class=\\"search-facets-group-header enabled\\"><i class=\\"md-icon material-icons\\">keyboard_arrow_down</i><span>filters.source.value</span><div class=\\"search-facets-group-remove-filters\\" title=\\"filters.reset.value\\"><i class=\\"md-icon material-icons\\">close</i></div></div><div class=\\"search-facets-group-items\\"><div class=\\"search-facets-group-item\\"><div class=\\"md-selection-control-container checkbox-green\\"><input type=\\"checkbox\\" id=\\"searchFacetsFeedsGroupItemCheckbox-focus-focus-id-1\\" checked=\\"\\" class=\\"md-selection-control-input\\" name=\\"searchFacetsFeedsGroupItemCheckboxs-focus-focus-id-1[]\\" aria-hidden=\\"true\\"/><label for=\\"searchFacetsFeedsGroupItemCheckbox-focus-focus-id-1\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"true\\" aria-labelledby=\\"\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--theme-secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box</i></div></label></div><div id=\\"searchFacetsFeedsGroupItemName-focus-focus-id-1\\" class=\\"search-facets-group-item-name\\">filters.source.focus: focus-name-1</div></div><div class=\\"search-facets-group-item feed-item\\"><div class=\\"md-selection-control-container checkbox-green\\"><input type=\\"checkbox\\" id=\\"searchFacetsFeedsGroupItemCheckbox-feed-feed-id-1\\" checked=\\"\\" class=\\"md-selection-control-input\\" name=\\"searchFacetsFeedsGroupItemCheckboxs-feed-feed-id-1[]\\" aria-hidden=\\"true\\"/><label for=\\"searchFacetsFeedsGroupItemCheckbox-feed-feed-id-1\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"true\\" aria-labelledby=\\"\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--theme-secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box</i></div></label></div><i class=\\"md-icon icon-news\\"></i><div id=\\"searchFacetsFeedsGroupItemName-feed-feed-id-1\\" class=\\"search-facets-group-item-name\\">feed-name-1</div></div><div class=\\"search-facets-group-item feed-item\\"><div class=\\"md-selection-control-container checkbox-green\\"><input type=\\"checkbox\\" id=\\"searchFacetsFeedsGroupItemCheckbox-feed-feed-id-2\\" checked=\\"\\" class=\\"md-selection-control-input\\" name=\\"searchFacetsFeedsGroupItemCheckboxs-feed-feed-id-2[]\\" aria-hidden=\\"true\\"/><label for=\\"searchFacetsFeedsGroupItemCheckbox-feed-feed-id-2\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"true\\" aria-labelledby=\\"\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--theme-secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box</i></div></label></div><i class=\\"md-icon icon-socialmedia\\"></i><div id=\\"searchFacetsFeedsGroupItemName-feed-feed-id-2\\" class=\\"search-facets-group-item-name\\">feed-name-2</div></div><div class=\\"search-facets-group-item feed-item\\"><div class=\\"md-selection-control-container checkbox-green\\"><input type=\\"checkbox\\" id=\\"searchFacetsFeedsGroupItemCheckbox-feed-feed-id-3\\" checked=\\"\\" class=\\"md-selection-control-input\\" name=\\"searchFacetsFeedsGroupItemCheckboxs-feed-feed-id-3[]\\" aria-hidden=\\"true\\"/><label for=\\"searchFacetsFeedsGroupItemCheckbox-feed-feed-id-3\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"true\\" aria-labelledby=\\"\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--theme-secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box</i></div></label></div><i class=\\"md-icon icon-print\\"></i><div id=\\"searchFacetsFeedsGroupItemName-feed-feed-id-3\\" class=\\"search-facets-group-item-name\\">feed-name-3</div></div></div></div>"`;

exports[`<SearchFacetsFeedsGroup /> Test component snapshot with focus list not loaded 1`] = `null`;

exports[`<SearchFacetsFeedsGroup /> Test component snapshot with group closed 1`] = `"<div class=\\"search-facets-group \\" id=\\"searchFacetsFeedsGroup-focus\\"><div class=\\"search-facets-group-header \\"><i class=\\"md-icon material-icons\\">keyboard_arrow_right</i><span>filters.source.value</span></div></div>"`;

exports[`<SearchFacetsFeedsGroup /> Test component snapshot with group disabled (no focus) 1`] = `"<div class=\\"search-facets-group disabled\\" id=\\"searchFacetsFeedsGroup-focus\\"><div class=\\"search-facets-group-header \\"><i class=\\"md-icon material-icons\\">keyboard_arrow_right</i><span>filters.source.value</span></div></div>"`;

exports[`<SearchFacetsFeedsGroup /> Test component snapshot with group opened and no filters enabled 1`] = `"<div class=\\"search-facets-group \\" id=\\"searchFacetsFeedsGroup-focus\\"><div class=\\"search-facets-group-header \\"><i class=\\"md-icon material-icons\\">keyboard_arrow_down</i><span>filters.source.value</span></div><div class=\\"search-facets-group-items\\"><div class=\\"search-facets-group-item\\"><div class=\\"md-selection-control-container\\"><input type=\\"checkbox\\" id=\\"searchFacetsFeedsGroupItemCheckbox-focus-focus-id-1\\" class=\\"md-selection-control-input\\" name=\\"searchFacetsFeedsGroupItemCheckboxs-focus-focus-id-1[]\\" aria-hidden=\\"true\\"/><label for=\\"searchFacetsFeedsGroupItemCheckbox-focus-focus-id-1\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"false\\" aria-labelledby=\\"\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box_outline_blank</i></div></label></div><div id=\\"searchFacetsFeedsGroupItemName-focus-focus-id-1\\" class=\\"search-facets-group-item-name\\">filters.source.focus: focus-name-1</div></div><div class=\\"search-facets-group-item feed-item\\"><div class=\\"md-selection-control-container\\"><input type=\\"checkbox\\" id=\\"searchFacetsFeedsGroupItemCheckbox-feed-feed-id-1\\" class=\\"md-selection-control-input\\" name=\\"searchFacetsFeedsGroupItemCheckboxs-feed-feed-id-1[]\\" aria-hidden=\\"true\\"/><label for=\\"searchFacetsFeedsGroupItemCheckbox-feed-feed-id-1\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"false\\" aria-labelledby=\\"\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box_outline_blank</i></div></label></div><i class=\\"md-icon icon-news\\"></i><div id=\\"searchFacetsFeedsGroupItemName-feed-feed-id-1\\" class=\\"search-facets-group-item-name\\">feed-name-1</div></div><div class=\\"search-facets-group-item feed-item\\"><div class=\\"md-selection-control-container\\"><input type=\\"checkbox\\" id=\\"searchFacetsFeedsGroupItemCheckbox-feed-feed-id-2\\" class=\\"md-selection-control-input\\" name=\\"searchFacetsFeedsGroupItemCheckboxs-feed-feed-id-2[]\\" aria-hidden=\\"true\\"/><label for=\\"searchFacetsFeedsGroupItemCheckbox-feed-feed-id-2\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"false\\" aria-labelledby=\\"\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box_outline_blank</i></div></label></div><i class=\\"md-icon icon-socialmedia\\"></i><div id=\\"searchFacetsFeedsGroupItemName-feed-feed-id-2\\" class=\\"search-facets-group-item-name\\">feed-name-2</div></div><div class=\\"search-facets-group-item feed-item\\"><div class=\\"md-selection-control-container\\"><input type=\\"checkbox\\" id=\\"searchFacetsFeedsGroupItemCheckbox-feed-feed-id-3\\" class=\\"md-selection-control-input\\" name=\\"searchFacetsFeedsGroupItemCheckboxs-feed-feed-id-3[]\\" aria-hidden=\\"true\\"/><label for=\\"searchFacetsFeedsGroupItemCheckbox-feed-feed-id-3\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"false\\" aria-labelledby=\\"\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box_outline_blank</i></div></label></div><i class=\\"md-icon icon-print\\"></i><div id=\\"searchFacetsFeedsGroupItemName-feed-feed-id-3\\" class=\\"search-facets-group-item-name\\">feed-name-3</div></div></div></div>"`;
